docs/guides/scheduler_event.md

```diff
@@ -0,0 +1,107 @@
+
+## How to use it
+
+To usage of the scheduler is easy, take a look at the example below of printing the raw JSON from Discord of the MESSAGE_CREATE event and making Concord ignore the event:
+
+```c
+enum discord_event_scheduler scheduler(struct discord *client, const char data[], size_t length, enum discord_gateway_events event) {
+    /* The "event" parameter is the parameter allows you to distinguish the events, and the data one is the JSON Discord sent. */
+    if (event == DISCORD_EV_MESSAGE_CREATE) {
+        log_debug("Received event JSON: %s", data);
+        /* Returning the "event ignore" enum, that says to Concord to ignore the event. */
+        return DISCORD_EVENT_IGNORE;
+    }
+    /* Here we are "saying" to Concord to handle the rest of the received events. */
+    return DISCORD_EVENT_MAIN_THREAD;
+    /* The "length" parameter can be used to parse the JSON, you can use jsmnf, from müller, in case you want to parse separated fields, or you can use discord_xxx_from_json() functions, in case you want to parse the whole JSON. */
+}
+```
+
+And this is it, but don't forget to add `discord_set_event_scheduler(client, scheduler);` to your code, otherwise, Concord will not use the scheduler.
+
+## How to parse the JSON (discord_xxx_from_json)
+
+This method is by far the easiest one, but it is not the fastest one, and you will need to free the returned struct, but this won't be a problem for most cases, so don't worry.
+
+`discord_xxx_from_json` isn't a single function, it's a set of functions, one for each event, so you will need to use the correct one, for example, if you want to parse a `message create` event, you will need to use `discord_message_from_json`, but never forget to free the returned struct, otherwise you will have a memory leak.
+
+Take an example of parsing a `message create` event using scheduler:
+
+```c
+enum discord_event_scheduler scheduler(struct discord *client, const char data[], size_t length, enum discord_gateway_events event) {
+    if (event == DISCORD_EV_MESSAGE_CREATE) {
+        struct discord_message self;
+        /* Parse the JSON and store the it in the "self" struct. */
+        discord_message_from_json(data, length, &self);
+
+        log_debug("Message content: %s", self.content);
+
+        /* Free the struct. */
+        discord_message_cleanup(&self);
+        return DISCORD_EVENT_IGNORE;
+    }
+    return DISCORD_EVENT_MAIN_THREAD;
+}
+```
+
+Mhm, that's easy! Now, if you want to parse a single JSON key, you should see the next section.
+
+## How to parse the JSON (jsmnf)
+
+This method is a bit more complicated, but it is the fastest one, and you will not need to free the returned struct, so it's a win-win situation.
+
+Note that the `jsmn-find` library, the one we are going to use, it's the same which Concord uses for parsing.
+
+Take an example of parsing a message content from `message create` event with scheduler:
```

Comment:
```suggestion
This method trades simplicity for speed, but the returned structure does not have to be freed.

Note that the `jsmn-find` library, the one we are going to use, it's the same which Concord uses for parsing.

For example, a developer could parse message content from `message_create` with the following:
```
Personally, I'd keep phrases that arguable don't contribute to the guide out such as "it's a win-win situation." I also shortened some phrases that do contribute to let it flow with the guide.

Reply:
I would agree with the first one, but in the second I don't see any reason to add "a developer could parse", and saying "Take an example" is way better